spinalcordtoolbox/scripts/sct_compute_compression.py

```diff
@@ -643,7 +645,7 @@ def main(argv: Sequence[str]):
             metric_ratio_PAM50_result = None
             metric_ratio_norm_result = None
 
-        rows.append([arguments.i, level, slice_compressed[idx],
+        rows.append([arguments.i, level, slice,
```

Comment:
Just so I understand the source of the error:

- `slice_compressed` is a list of slices where the compression exists:
  https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/3c50bc2380f2b953e8b60b96233b557b83fb6bbe/spinalcordtoolbox/scripts/sct_compute_compression.py#L576-L577
- `idx` comes from `compressed_levels_dict`:
  https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/3c50bc2380f2b953e8b60b96233b557b83fb6bbe/spinalcordtoolbox/scripts/sct_compute_compression.py#L621
- And `compressed_levels_dict` comes from `slice_compressed`:
  https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/3c50bc2380f2b953e8b60b96233b557b83fb6bbe/spinalcordtoolbox/scripts/sct_compute_compression.py#L578
  This is weird. If both `idx` and `slice_compressed` come from the same source, then why does `slice_compressed[idx]` produce the wrong slice numbers?

----

Answer: `slice_compressed` is sorted by `z` slice index (I->S):

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/3c50bc2380f2b953e8b60b96233b557b83fb6bbe/spinalcordtoolbox/scripts/sct_compute_compression.py#L174

While `compressed_levels_dict` is sorted by vertebral level order (S->I):

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/3c50bc2380f2b953e8b60b96233b557b83fb6bbe/spinalcordtoolbox/scripts/sct_compute_compression.py#L212

Reply:
It's a bit weird that `compressed_levels_dict` is sorted S->I while everything else is sorted I->S, so maybe that's worth changing?

But, regardless, I like your solution, because you explicitly get the slice number from the same source as the level, which prevents order from having any effect at all.

Reply:
> It's a bit weird that `compressed_levels_dict` is sorted S->I while everything else is sorted I->S, so maybe that's worth changing?

One reason to keep the order `I->S` is that the CSV table would be more accurate. Right now, the CSV is sorted S->I, yet the header says I->S, which might be a bit confusing?

Reply:
Thanks for digging! 

> Right now, the CSV is sorted S->I, yet the header says I->S, which might be a bit confusing?

Yeah, this is confusing! An easy fix would be to update the header from `slice(I->S)` to `slice(S->I)`. But, by this, we would introduce an inconsistency with a CSV file produced by `sct_process_segmentation`, which uses `Slice (I->S)`:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/264760c139548190650ecce8271b19bd5425a46e/spinalcordtoolbox/aggregate_slicewise.py#L559

So maybe a solution might be to reverse the table as all:

filename | compression_level | slice(I->S) | diameter_AP_ratio |   |  
-- | -- | -- | -- | -- | --
t2_compressed_seg.nii.gz | 6.0 | 99 | 21.88085837700506 |   |  
t2_compressed_seg.nii.gz | 5.0 | 118 | 10.780650030491945 |   |  
t2_compressed_seg.nii.gz | 4.0 | 140 | 11.314949666366614 |   |  
t2_compressed_seg.nii.gz | 3.0 | 159 | 3.3493565531077407 |   |  

Reply:
nitpick: we can rename `slice(I->S)` to `Slice (I->S)` to be consistent with https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/264760c139548190650ecce8271b19bd5425a46e/spinalcordtoolbox/aggregate_slicewise.py#L559

Reply:
> So maybe a solution might be to reverse the table as all:

I think so too! I think we can accomplish this by not inverting the order (as done above)?

Reply:
- `slice(I->S)` renamed to `Slice (I->S)` in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4326/commits/fbb1b70ee61b8e562c0bdb6bbcefeb9ac70d23bf
- reverse rows in the output CSV table to match `Slice (I->S)` ordering in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4326/commits/a1df9bd2c26b81ca391721c8fda03d1d68b2e8ee
<img width="700" alt="image" src="https://github.com/spinalcordtoolbox/spinalcordtoolbox/assets/39456460/af508917-d00e-415b-9724-04b955e0dac2">

Reply:
hmm, I don't remember if it was to have compression levels in order, or if there was more!